src/internal.rs

```diff
@@ -322,57 +350,64 @@ impl Rav1dFrameContext_bd_fn {
 
 #[repr(C)]
 pub struct CodedBlockInfo {
-    pub eob: [i16; 3],
-    pub txtp: [u8; 3],
+    pub eob: [i16; 3 /* plane */],
+    pub txtp: [u8; 3 /* plane */],
 }
 
 #[repr(C)]
 pub struct Rav1dFrameContext_frame_thread {
-    pub next_tile_row: [c_int; 2],
+    pub next_tile_row: [c_int; 2 /* 0: reconstruction, 1: entropy */],
     pub entropy_progress: atomic_int,
-    pub deblock_progress: atomic_int,
+    pub deblock_progress: atomic_int, // in sby units
     pub frame_progress: *mut atomic_uint,
     pub copy_lpf_progress: *mut atomic_uint,
+    // indexed using t->by * f->b4_stride + t->bx
     pub b: *mut Av1Block,
     pub cbi: *mut CodedBlockInfo,
-    pub pal: *mut [[u16; 8]; 3],
+    // indexed using (t->by >> 1) * (f->b4_stride >> 1) + (t->bx >> 1)
+    pub pal: *mut [[u16; 8 /* idx */]; 3 /* plane */],
+    // iterated over inside tile state
     pub pal_idx: *mut u8,
     pub cf: *mut DynCoef,
     pub prog_sz: c_int,
     pub pal_sz: c_int,
     pub pal_idx_sz: c_int,
     pub cf_sz: c_int,
+    // start offsets per tile
     pub tile_start_off: *mut c_int,
 }
 
+/// loopfilter
 #[repr(C)]
 pub struct Rav1dFrameContext_lf {
     pub level: *mut [u8; 4],
     pub mask: *mut Av1Filter,
     pub lr_mask: *mut Av1Restoration,
-    pub mask_sz: c_int,
+    pub mask_sz: c_int, /* w*h */
     pub lr_mask_sz: c_int,
-    pub cdef_buf_plane_sz: [c_int; 2],
+    pub cdef_buf_plane_sz: [c_int; 2], /* stride*sbh*4 */
     pub cdef_buf_sbh: c_int,
-    pub lr_buf_plane_sz: [c_int; 2],
-    pub re_sz: c_int,
+    pub lr_buf_plane_sz: [c_int; 2], /* (stride*sbh*4) << sb128 if n_tc > 1, else stride*4 */
+    pub re_sz: c_int, /* h */
     pub lim_lut: Align16<Av1FilterLUT>,
     pub last_sharpness: c_int,
-    pub lvl: [[[[u8; 2]; 8]; 4]; 8],
+    pub lvl: [[[[u8; 2 /* is_gmv */]; 8 /* ref */]; 4 /* dir */]; 8 /* seg_id */],
     pub tx_lpf_right_edge: [*mut u8; 2],
     pub cdef_line_buf: *mut u8,
     pub lr_line_buf: *mut u8,
-    pub cdef_line: [[*mut DynPixel; 3]; 2],
-    pub cdef_lpf_line: [*mut DynPixel; 3],
-    pub lr_lpf_line: [*mut DynPixel; 3],
+    pub cdef_line: [[*mut DynPixel; 3 /* plane */]; 2 /* pre, post */],
+    pub cdef_lpf_line: [*mut DynPixel; 3 /* plane */],
+    pub lr_lpf_line: [*mut DynPixel; 3 /* plane */],
+
+    // in-loop filter per-frame state keeping
     pub start_of_tile_row: *mut u8,
     pub start_of_tile_row_sz: c_int,
     pub need_cdef_lpf_copy: c_int,
     pub p: [*mut DynPixel; 3],
     pub sr_p: [*mut DynPixel; 3],
     pub mask_ptr: *mut Av1Filter,
     pub prev_mask_ptr: *mut Av1Filter,
-    pub restore_planes: c_int,
+    pub restore_planes: c_int, // enum LrRestorePlanes
```

Comment:
```suggestion
    pub restore_planes: LrRestorePlanes,
```

We should be able to just use the `enum` directly.  C forces `enum`s to be `int`-sized, but Rust doesn't (though in this case, it's still `c_int`, but future `dav1d` versions could change that, as they sometimes make the type `uint8_t` to save space).

Reply:
Sure, but let's do that in a separate change later.